addons/stock/models/stock_picking.py

```diff
@@ -513,7 +513,8 @@ def write(self, vals):
             after_vals['location_dest_id'] = vals['location_dest_id']
         if after_vals:
             self.mapped('move_lines').filtered(lambda move: not move.scrapped).write(after_vals)
-        if vals.get('move_lines'):
+        # avoid to autoconfirm draft planned transfer if the context is lost.
+        if vals.get('move_lines') and (self.state != 'draft' or all(ml.product_uom_qty == 0 for ml in self.mapped('move_lines'))):
```

Comment:
# Do not run autoconfirm if any of the moves have an initial demand. If they have
# an initial demand, they were created through the "planned transfer" mechanism.

Reply:
not any(self.mapped('move_lines.product_uom_qty') ?
not sure it's cleaner